lib/BlockedBy.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const NodeResque = require('node-resque');
+const { queuePrefix } = require('../config/redis');
+
+class BlockedBy extends NodeResque.Plugin {
+    async beforePerform() {
+        const { jobId, blockedBy } = await this.jobInfo();
+        const blockingJobKeys = await this.queueObject.connection.redis.mget(blockedBy);
```

Comment:
Doesn't `blockedBy` have stuff like `~sd@NUM`? When you set it below, you just set the key based on the jobID, so how would mget return anything here? Or maybe I'm missing something.

Reply:
That logic is taken care of in the models: https://github.com/screwdriver-cd/models/blob/f0b2a8291e8982aacab885d66d91a53ba3e8be3d/lib/build.js#L51-L70